Mathlib/Data/Fintype/Quotient.lean

```diff
@@ -50,24 +50,26 @@ theorem listChoice_mk {l : List ι} (a : ∀ i ∈ l, α i) : listChoice (S := S
     rw [listChoice_mk]
     exact congrArg (⟦·⟧) (List.Pi.cons_eta a)
 
+omit [DecidableEq ι] in
 /-- Choice-free induction principle for quotients indexed by a `List`. -/
 @[elab_as_elim]
 lemma list_ind {l : List ι} {C : (∀ i ∈ l, Quotient (S i)) → Prop}
     (f : ∀ a : ∀ i ∈ l, α i, C (⟦a · ·⟧)) (q : ∀ i ∈ l, Quotient (S i)) : C q :=
   match l with
   |     [] => cast (congr_arg _ (funext₂ nofun)) (f nofun)
   | i :: l => by
-    rw [← List.Pi.cons_eta q]
-    induction' List.Pi.head q using Quotient.ind with a
-    refine @list_ind _ (fun q ↦ C (List.Pi.cons _ _ ⟦a⟧ q)) ?_ (List.Pi.tail q)
-    intro as
-    rw [List.Pi.cons_map a as (fun i ↦ Quotient.mk (S i))]
-    exact f _
+    classical
+      rw [← List.Pi.cons_eta q]
+      induction' List.Pi.head q using Quotient.ind with a
+      refine @list_ind _ (fun q ↦ C (List.Pi.cons _ _ ⟦a⟧ q)) ?_ (List.Pi.tail q)
+      intro as
+      rw [List.Pi.cons_map a as (fun i ↦ Quotient.mk (S i))]
+      exact f _
```

Comment:
```suggestion
    rw [← List.Pi.cons_eta q]
    induction' List.Pi.head q using Quotient.ind with a
    refine @list_ind _ (fun q ↦ C (List.Pi.cons _ _ ⟦a⟧ q)) ?_ (List.Pi.tail q)
    intro as
    rw [List.Pi.cons_map a as (fun i ↦ Quotient.mk (S i))]
    exact f _
```
I think the current convention is not to indent further after classical

Reply:
@FR-vdash-bot You've added this lemma. Do you care about it being choice free? What do you prefer to make the pending linter happy:

- make it choice dependent as most of `Mathlib`;
- add a choice free version in `Decidable` namespace?

Reply:
I'm going to revert this file for now.